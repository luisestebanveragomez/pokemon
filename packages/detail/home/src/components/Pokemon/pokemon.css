.contFigure {
    display: grid;
    justify-content: center;
}

.figure__pokemon {
    display: grid;
    justify-content: center;
    width: 100%;
    z-index: 2;
    gap: 20px;
    grid-template-columns: 1fr;
}

.figure__pokemon-img {
    height: auto;
    width: 200px;
}

.figure__pokemon-figcaption {
    margin-top: 2rem;
    text-align: center;
}

.figure__pokemon-name {
    font-style: normal;
    font-weight: bold;
    font-size: 3rem;
    line-height: 60px;
    color: var(--white);
    text-transform: capitalize;
}

.figure__pokemon-number {
    font-size: 15vmax;
    color: rgba(255, 255, 255, 0.18);
    user-select: none;
    right: 0;
    top: 20%;
    z-index: 1;
    font-weight: 900;
    text-align: center;
    left: 0;
}

@media screen and (min-width: 800px) {
    .figure__pokemon {
        grid-template-columns: 1fr 1fr;
        gap: 200px;
    }
    .figure__pokemon-number {
        top: 50%;
        transform: translateY(-50%);
    }
}