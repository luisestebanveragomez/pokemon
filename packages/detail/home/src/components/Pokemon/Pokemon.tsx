import { FC } from "react";

import "./pokemon.css";

interface PokemonProps {
  name: string;
  id: number | string;
  textButton: string;
  handle: (id: string | number) => void;
}

export const Pokemon: FC<PokemonProps> = ({ name, id, textButton, handle }) => {
  return (
    <article className="relative contFigure">
      <figure className="relative figure__pokemon">
        <img
          className="figure__pokemon-img"
          src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${id}.svg`}
          alt={`pokemon ${name}`}
          loading="lazy"
          title={name}
        />
        <figcaption className="figure__pokemon-figcaption">
          <h2 className="figure__pokemon-name">{name}</h2>
          <button
            className="generic__button"
            type="button"
            onClick={() => handle(id)}
          >
            {textButton}
          </button>
        </figcaption>
      </figure>
      <div className="figure__pokemon-number absolute">#{id}</div>
    </article>
  );
};
