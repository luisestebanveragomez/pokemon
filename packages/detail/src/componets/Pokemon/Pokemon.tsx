import { FC } from "react";

import "./pokemon.css";
import { Pokemon } from "../../models/pokemonDetail.model";

interface PokemonProps {
  detail: Pokemon;
  id: string;
}

export const SectionPokemon: FC<PokemonProps> = ({ detail, id }) => {
  return (
    <section className="relative">
      <figure className="pokemon">
        <img
          src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${id}.svg`}
          className="pokemon__img"
          alt=""
        />
        <figcaption>
          <span className="pokemon__category">{detail.type}</span>
          <h2 className="pokemon__name">{detail.name}</h2>
        </figcaption>
      </figure>
      <div className="pokemon__number">#{id}</div>
    </section>
  );
};
