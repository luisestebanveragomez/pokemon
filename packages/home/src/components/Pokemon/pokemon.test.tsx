import { fireEvent, render } from "@testing-library/react";
import { Pokemon } from "./Pokemon";

describe("SectionPokemon Component", () => {
  it("Should Render", () => {
    const { container } = render(
      <Pokemon name={"test"} id={12} textButton={""} handle={() => {}} />
    );
    expect(container).toMatchSnapshot();
  });

  it("Should Return Id", () => {
    const id = 123;
    let result;
    const handleEvent = (val: number) => (result = val);
    const { container } = render(
      <Pokemon name={"test"} id={id} textButton={""} handle={handleEvent} />
    );
    const button = container.querySelector("button");
    fireEvent.click(button);
    expect(result).toEqual(id);
  });
});
