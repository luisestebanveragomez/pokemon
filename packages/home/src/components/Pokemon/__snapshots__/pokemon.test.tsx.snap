// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Component Should Render 1`] = `
<div>
  <article
    class="relative contFigure"
  >
    <figure
      class="relative figure__pokemon"
    >
      <img
        alt="pokemon test"
        class="figure__pokemon-img"
        loading="lazy"
        src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/12.svg"
        title="test"
      />
      <figcaption
        class="figure__pokemon-figcaption"
      >
        <h2
          class="figure__pokemon-name"
        >
          test
        </h2>
        <button
          class="generic__button"
          type="button"
        />
      </figcaption>
    </figure>
    <div
      class="figure__pokemon-number absolute"
    >
      #
      12
    </div>
  </article>
</div>
`;
